src/source.rs

```diff
@@ -20,10 +20,15 @@ use std::fmt::{Display, Formatter};
 use std::fs::OpenOptions;
 use std::io::{Read, Seek, SeekFrom};
 use std::path::Path;
+use std::time::Duration;
 
 use crate::cmdline::*;
 use crate::errors::*;
 
+/// Default timeout for HTTP requests (4 hours).
+const DEFAULT_HTTP_REQ_TIMEOUT: Duration = Duration::from_secs(4 * 60 * 60);
```

Comment:
Let's not call this the "default" since it's not configurable.

Reply:
Ack, I've renamed to `HTTP_COMPLETION_TIMEOUT`. Were you hinting at keeping a ticket around to maybe later add a CLI flag to configure this, or do you think it isn't useful to have that?

Reply:
Nah, I don't think it'd be useful.  We can always add later if needed.